accepted/0000-caasp-integration.md

```diff
@@ -0,0 +1,298 @@
+    require:
+      - pkg: mgr_sshd_installed
+  ```
+
+  Finally, SUSE Manager copies the templates for deployment files provided by `skuba` (in `/usr/share/caasp/AutoYaST/bare-metal/AutoYaST.xml`) from the Management Node to the SUSE Manager itself.
+
+  Given that the Management Node is a Salt minion managed by SUSE Manager, every update for `SUSE-CaaSP-Management` can be installed directly from SUSE Manager. Extra care should be put into protecting the cluster configuration file generated by `skuba` during cluster bootstrapping (rsync into SUSE Manager nightly?).
+
+  ##### Load balancer
+
+  Every CaaS Platform production cluster must have at least one load balancer associated. There is not a standardized way to deploy a load balancer. One example that the CaaS Platform suggests using is the [`nginx` TCP Load Balancer with passive checks](https://susedoc.github.io/doc-caasp/beta/caasp-deployment/single-html/#_configuring_the_load_balancer).
+
+  In this case, every Salt minion registered with SUSE Manager that runs SUSE Linux Enterprise 15 or OpenSUSE can be used to be a load balancer. The configuration of the load balancer can be offered via a new "CaaS Platform" entitlement that modifies the highstate of the minion by configuring `nginx` as the load balancer.
+
+  Every update to the load balancer `nginx` will be handled via SUSE Manager.
+
+  ##### Bootstrapping an empty cluster
+
+  Salt will invoke `skuba` (using `cmd.run`) in the Management Node. Unfortunately, `skuba` does not offer an API yet (is in the progress).
+  Example:
+
+  ```
+  bootstrap_master:
+    cmd.run:
+      - name: skuba cluster init --control-plane <load balancer IP> <cluster name>
+  ```
+
+  The parameters above are already known to SUSE Manager.
+
+  It is probably not in the scope of this RFC to offer to the user the ability to customize the configuration directly from SUSE Manager, but it will be in the future. At this point, the configuration of the cluster is in the Management Node and can be customized by the user (e.g. [integrating an external LDAP](https://susedoc.github.io/doc-caasp/beta/caasp-deployment/single-html/#_integrate_external_ldap) or [configure `kured`](https://susedoc.github.io/doc-caasp/beta/caasp-deployment/single-html/#_prevent_nodes_running_special_workloads_from_being_rebooted)). 
+  
+  Single Sign-On is out of scope until further developments from the Product Management.
+  
+  ##### Master and worker nodes
+
+  ###### Registering master and worker to SUSE Manager
+
+  The core of the CaaS Platform is master and worker nodes. The Management Node, instructed by SUSE Manager, will take care of bootstrapping master and worker nodes.
+
+  The critical part of every master and worker node is handling updates of the underlying Kubernetes-related packages (`kubernetes-kubeadm kubernetes-kubelet kubernetes-client cri-o cni-plugins`): these updates must be handled with `skuba` from the Management Node.
```

Comment:
Not strictly from the management node, as in this case upgrades do not rely on information present in the "cluster definition folder", so upgrades could be triggered from anywhere, as long as they can SSH into the nodes.

Reply:
Then maybe we should keep the nodes registered in SUMA, possibly with some special restrictions via new types of entitlements.

We can also have some specialized view for cluster machines just like "Systems > Virtual Systems" page.

Reply:
> Not strictly from the management node, as in this case upgrades do not rely on information present in the "cluster definition folder", so upgrades could be triggered from anywhere, as long as they can SSH into the nodes.

From a SUSE Manager point of view, the only registered system that can trigger `skuba` is the Management Node. Outside of that, any user with `skuba` and the correct deployed SSH keys can trigger the updates. But this special case would be outside from the RFC, wouldn't it?

Reply:
> Then maybe we should keep the nodes registered in SUMA, possibly with some special restrictions via new types of entitlements.
> 
> We can also have some specialized view for cluster machines just like "Systems > Virtual Systems" page.

Do we already have special entitlements that package locks systems?
I know that we have package locks for traditional system, but not for Salt systems.